nucypher/blockchain/eth/agents.py

```diff
@@ -1122,3 +1124,90 @@ def dump(self) -> list:
             entry = self.contract.functions.seeds(ip).call()
             entries.append(entry)
         return entries
+
+
+class MultiSigAgent(EthereumContractAgent):
+
+    registry_contract_name = MULTISIG_CONTRACT_NAME
+
+    Vector = List[str]
+
+    @property
+    def nonce(self) -> int:
+        nonce = self.contract.functions.nonce().call()
+        return nonce
+
+    def get_owner(self, index: int) -> str:
+        owner = self.contract.functions.owners(index).call()
+        return owner
+
+    @property
+    def owners(self) -> Tuple[str]:
+        i = 0
+        owners = list()
+        array_is_within_bounds = True
+        while array_is_within_bounds:
+            try:
+                owner = self.get_owner(i)
+            except (TransactionFailed, ValueError):
+                array_is_within_bounds = False
+            else:
+                owners.append(owner)
+                i += 1
+        return tuple(owners)
+
+    @property
+    def threshold(self) -> int:
+        threshold = self.contract.functions.required().call()
+        return threshold
+
+    @validate_checksum_address
+    def is_owner(self, checksum_address: str) -> bool:
+        result = self.contract.functions.isOwner(checksum_address).call()
+        return result
+
+    def add_owner(self, new_owner_address: str, sender_address: str) -> dict:
```

Comment:
Should we validate checksum address here? And  on Line 1182, and 1198?

Reply:
Let's allow for this in a follow up PR.